File: Python/Algorithms/Searching_Algorithms/linear_Search.py
import sys 


'''  variable assignment =

    -n is size of array
    -s_element is element to be searched in the array
''' 


# The function returns location of s_element in given array arr 
# if present, else returns -1 

def linear_search(arr,s_element):
	n=len(arr)
	for i in range(n): #comparing each element with s_element and if found then return its index
		if arr[i]==s_element:
			return i
	return -1

# Driver code

if __name__ == '__main__':
	print('enter the number of elements')
	n=int(input())

	print('\nenter the elements in a single line with space')

	arr=list(map(int,input().split()))

	print('\nenter the element to be searched')
	s=int(input())

	result=linear_search(arr,s) #calling function

	print("Number" , s, "is at index" ,"%.0f"%result)

File: C++/Algorithms/Searching-Algorithms/LinearSearch.cpp
// C++ code to linearly search x in arr[]. If x 
// is present then return its location, otherwise 
// return -1 

#include <iostream> 
using namespace std; 

int search(int arr[], int n, int y) 
{ 
	int i; 
	for (i = 0; i < n; i++) 
		if (arr[i] == y) 
			return i; 
	return -1; 
} 

int main(void) 
{ 
	int arr[] = { 2, 3, 4, 10, 40 }; 
	int y = 10; 
	int n = sizeof(arr) / sizeof(arr[0]); 
	int result = search(arr, n, y); 
(result == -1)? cout<<"Element is not present in array"
				: cout<<"Element is present at index " <<result; 
return 0; 
} 
